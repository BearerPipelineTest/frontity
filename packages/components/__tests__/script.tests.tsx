import { act, create } from "react-test-renderer";
import Script from "../script";

describe("Script", () => {
  test("should not render a script tag without code or src", () => {
    let root;
    act(() => {
      root = create(<Script id="script-1">{"let index = 0;"}</Script>);
    });

    expect(root.toJSON()).toMatchSnapshot();
    expect(document.body.querySelector("#script-1")).toBeNull();
  });

  test("should render a script tag with src", () => {
    let root;
    act(() => {
      root = create(<Script id="script-2" src="source" />);
    });

    expect(root.toJSON()).toMatchSnapshot();
    expect(document.body.querySelector("#script-2")).toMatchInlineSnapshot(`
      <script
        id="script-2"
        src="source"
      />
    `);
  });

  test("should not render the passed children with src", () => {
    let root;
    act(() => {
      root = create(
        <Script
          id="script-3"
          src="source-with-children"
        >{`let children = null;`}</Script>
      );
    });

    expect(root.toJSON()).toMatchSnapshot();
    expect(document.body.querySelector("#script-3")).toMatchInlineSnapshot(`
      <script
        id="script-3"
        src="source-with-children"
      />
    `);
  });

  test("should execute the code", () => {
    let root;
    act(() => {
      root = create(<Script id="script-4" code="global.__i__ = Infinity;" />);
    });

    expect(root.toJSON()).toMatchSnapshot();
    expect(document.body.querySelector("#script-4")).toBeNull();
    expect(global["__i__"]).toEqual(Infinity);
  });

  test("should remove the script tag at unmount", () => {
    let root;
    act(() => {
      root = create(<Script id="script-5" src="source-for-unmount" />);
    });

    expect(root.toJSON()).toMatchSnapshot();
    expect(document.body.querySelector("#script-5")).toMatchInlineSnapshot(`
      <script
        id="script-5"
        src="source-for-unmount"
      />
    `);

    act(() => {
      root.unmount();
    });

    expect(document.body.querySelector("#script-5")).toBeNull();
  });
});
