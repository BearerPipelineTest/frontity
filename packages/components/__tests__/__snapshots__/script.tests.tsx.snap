// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Script should execute the code 1`] = `null`;

exports[`Script should not render a script tag without code or src 1`] = `null`;

exports[`Script should not render the passed children with src 1`] = `null`;

exports[`Script should remove the script tag at unmount 1`] = `null`;

exports[`Script should render a script tag with src 1`] = `null`;
